### DISCUSSION TITLE
Mistral API backend

### DISCUSSION THREAD

**Discussion by hcal** [ggozad/oterm#42] [Ideas]
Oterm has quickly become my favorite interface for LLMs.  It has integrated itself into my workflow far more than I would have expected.  I know it is meant primarily as an UI for ollama, but it seems like it would be a good fit for using third party APIs as a model option too.  

I have a mistral API beta access.  If you're interested, I'd be willing to foot the bill for some initial testing.

**Comment by ggozad** [comment]
Hey there! It makes me really happy that others find it useful :)
Here are some thoughts: 
I picked Ollama because I am interested mostly in open-source models and I dislike using proprietary models. Ollama gives me a nice API to work with consistent regardless of the model I use with.
Having said that I would be interested to help you out adding support for mistral if you want to put the effort and make a PR.
It's about doing two things:
* API interface (see `oterm/ollama.py`)
* a different way of "embedding" the conversation. Ollama returns with each completion a "context" which essentially encompasses the whole conversation. I can save that in the db and restore it when you resume your chat. Mistral API does not do that, so it would have to happen differently, perhaps by including all past messages.

**Comment by ggozad** [comment]
Since @yilmaz08 updated how we interface with Ollama through the chat API, this is now possible, i.e. it would be great to add support for other services than Ollama, including Mistral.
This would entail having a common interface to talk with the client which is now easy, as well as UI changes to allow a choice in model providers with potentially separate configuration screens.
